crates/cheatnet/src/rpc.rs

```diff
@@ -649,6 +653,128 @@ pub fn call_contract_syscall(
+        syscall_handler.cheatcode_state,
+    )?;
+
+    let constructor_retdata = create_retdata_segment(
+        vm,
+        &mut syscall_handler.syscall_handler,
+        &call_info.execution.retdata.0,
+    )?;
+    update_remaining_gas(remaining_gas, &call_info);
+
+    syscall_handler.syscall_handler.inner_calls.push(call_info);
+
+    Ok(DeployResponse {
+        contract_address: deployed_contract_address,
+        constructor_retdata,
+    })
+}
+
+// blockifier/src/execution/execution_utils.rs:217 (execute_deployment)
+pub fn execute_deployment(
+    state: &mut dyn State,
+    resources: &mut ExecutionResources,
+    context: &mut EntryPointExecutionContext,
+    ctor_context: ConstructorContext,
+    constructor_calldata: Calldata,
+    remaining_gas: u64,
+    cheatcode_state: &CheatcodeState,
+) -> EntryPointExecutionResult<CallInfo> {
+    // Address allocation in the state is done before calling the constructor, so that it is
+    // visible from it.
+    let deployed_contract_address = ctor_context.storage_address;
+    let current_class_hash = state.get_class_hash_at(deployed_contract_address)?;
+    if current_class_hash != ClassHash::default() {
+        return Err(StateError::UnavailableContractAddress(deployed_contract_address).into());
+    }
+
+    state.set_class_hash_at(deployed_contract_address, ctor_context.class_hash)?;
+
+    let call_info = execute_constructor_entry_point(
+        state,
+        resources,
+        context,
+        ctor_context,
+        constructor_calldata,
+        remaining_gas,
+        cheatcode_state,
+    )?;
+
+    Ok(call_info)
+}
+// blockifier/src/execution/entry_point.rs:366 (execute_constructor_entry_point)
+pub fn execute_constructor_entry_point(
+    state: &mut dyn State,
+    resources: &mut ExecutionResources,
+    context: &mut EntryPointExecutionContext,
+    ctor_context: ConstructorContext,
+    calldata: Calldata,
+    remaining_gas: u64,
+    cheatcode_state: &CheatcodeState,
+) -> EntryPointExecutionResult<CallInfo> {
+    // Ensure the class is declared (by reading it).
+    let contract_class = state.get_compiled_contract_class(&ctor_context.class_hash)?;
+    let Some(constructor_selector) = contract_class.constructor_selector() else {
+        // Contract has no constructor.
+        return handle_empty_constructor(ctor_context, calldata, remaining_gas);
+    };
+
+    let mut constructor_call = CallEntryPoint {
+        class_hash: None,
+        code_address: ctor_context.code_address,
+        entry_point_type: EntryPointType::Constructor,
+        entry_point_selector: constructor_selector,
+        calldata,
+        storage_address: ctor_context.storage_address,
+        caller_address: ctor_context.caller_address,
+        call_type: CallType::Call,
+        initial_gas: remaining_gas,
+    };
+
+    execute_call_entry_point(
+        &mut constructor_call,
+        state,
+        cheatcode_state,
+        resources,
+        context,
+    )
+}
+
```

Comment:
Is it entirely copied?

Reply:
Good point. I'll mark the modified call

Reply:
[Mark modifed region](https://github.com/foundry-rs/starknet-foundry/pull/503/commits/d35aa372f6abeebf96f69384781a41004908bf23)